### DISCUSSION TITLE
Exporting data from grainBoundary objects?

### DISCUSSION THREAD

**Discussion by cmbean2** [mtex-toolbox/mtex#1729] [Ask Anything]
I have been doing some EBSD and grain boundary data analysis using the toolbox, and the tutorials have been very helpful, but I am hoping to export the figures or the raw data that have been created so that I can do further analysis using other tools. 

I am looking for help on either how to either export the created figure as a 32-bit tiff file so that the values are not rounded, as happens with the 'MTEX/Export Image' option that creates an 8-bit tiff.

Or I am looking for guidance on directly accessing the arrays stored in the 'grainBoundary' objects so that I can manually create the tiff files/plots I am interested in myself. I have not worked with these data objects before, and how the plot command uses these objects to create the resulting figure eludes me. 

The main command/data objects in question come from here: https://mtex-toolbox.github.io/BoundaryPlots.html
>>gB_Fo = grains.boundary('Fo','Fo');
>> plot(gB_Fo,gB_Fo.misorientation.angle./degree,'linewidth',4)

Any help would be appreciated thanks!

**Comment by kilir** [comment]
Hi,
to aces the data, you could do something like
```
mtexdata titanium

ebsd = ebsd.calcGrains;


gb = grains.boundary('t','t')

% face midpoints and misor angle
[gb.midPoint gb.misorientation.angle]

% or only the gb. vertices
gb.V
```
Hope that helps.

Cheers,
Rüdiger

**Reply by cmbean2** [reply]
Hi Rüdiger,

Ah guess I should clarify slightly; my bad. I understand how to get out the individual arrays within the object, but I am unsure which ones it uses to make the grain boundary map. 

The intensity value is clearly the misorientation.angle./degree, but which arrays are the x and y coordinates to tell the plot command to put the intensity in a certain location? As the arrays labeled x and y have fewer values than the rest of the arrays (e.g. misor has 39817x1 while x and y have only 39352x1) 

Ideally, I like to have an array that is (n by 3) where column 1 is the x coord, column 2 is the y coord and then I can fill the last in with the intensity (e.g. misor angle) or another calculated value.

Thanks for the help! I appreciate it!
Chris

**Reply by kilir** [reply]
Hi,
x and y are the vertices of a grain boundary segment and in most cases there are more vertices than segments. The property you assign to a grain boundary segment is plotted as the line between those vertices. That is why  `[gb.midPoint gb.misorientation.angle]` gives you an nx3 list. Vertices (gb.V) are indexed by `gb.F`. 

Since a vertex may serve three or more points across a boundary or a triple junction, it's not so easy to assign a single value to it - hence the property comes with the faces.

Hope that helps.

Cheers,
Rüdiger
